docs/content/sips/008-using-oci-registries.md

```diff
@@ -0,0 +1,298 @@
+asset, its OCI manifest, locked application, and resulting local directory
+structure when pulling the application locally.
+
+Consider the following `spin.toml` file for the application:
+
+```toml
+spin_version = "1"
+authors = ["Radu Matei <radu.matei@fermyon.com>"]
+description = ""
+name = "github-stars-webhook"
+trigger = { type = "http", base = "/" }
+version = "0.1.0"
+
+[[component]]
+id = "github-star-webhook"
+source = "target/spin-http-js.wasm"
+files = ["my-file.json"]
+allowed_http_hosts = ["https://hooks.slack.com"]
+[component.trigger]
+route = "/..."
+```
+
+Distributing it to the GitHub Container Registry:
+
+```bash
+$ spin oci push ghcr.io/radu-matei/spin-example:v1
+INFO spin_publish::oci::client: Pushed "https://ghcr.io/v2/radu-matei/spin-example/manifests/sha256:8f86a27fbc457416701c4d18680083f598076d0a52dca2a5936e92754a845ed1"
+
+$ spin oci pull ghcr.io/radu-matei/spin-example:v1
+INFO spin_publish::oci::client: Pulled ghcr.io/radu-matei/spin-example:v1@sha256:8f86a27fbc457416701c4d18680083f598076d0a52dca2a5936e92754a845ed1
+```
+
+This operation created the following local cache directory structure:
+
+```bash
+$ tree /Users/radu/Library/Application\ Support/fermyon/registry
+└── oci
+    ├── data
+    │   └── sha256:a4699e4f9ef3f4922f38f0d017aa26438908f38caf020a739e0ee27fe796eb02
+    ├── manifests
+    │   └── ghcr.io
+    │       └── radu-matei
+    │           └── spin-example
+    │               └── v1
+    │                   ├── config.json
+    │                   └── manifest.json
+    └── wasm
+        └── sha256:55c29ad4b0ad0c6bd8ec1ffc8f04e63342e5901280037ef706b1b114475d3cbb
+```
+
+Looking at `manifest.json`, we see the top-level media type for the artifact
+configuration to be `application/vnd.fermyon.spin.application.v1+config`
+
+```json
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.fermyon.spin.application.v1+config",
+    "digest": "sha256:b36160facea3076ad136c09bd4975a429805945ad313b4674363841d5a7f66a0",
+    "size": 643
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.wasm.content.layer.v1+wasm",
+      "digest": "sha256:55c29ad4b0ad0c6bd8ec1ffc8f04e63342e5901280037ef706b1b114475d3cbb",
+      "size": 2147122
+    },
+    {
+      "mediaType": "application/vnd.wasm.content.layer.v1+data",
+      "digest": "sha256:a4699e4f9ef3f4922f38f0d017aa26438908f38caf020a739e0ee27fe796eb02",
+      "size": 178
+    }
+  ]
+}
+```
+
+Note: the media types used are not final and *can* change based on the community
+standardization for Wasm modules in OCI registries.
+
+The manifest contains layers for the Wasm module for the component and for the
+one static asset referenced by the component. For each additional component, the
+Wasm module and static assets would be individual layers in the manifest above.
+
+Let's explore the OCI configuration object referenced in the manifest —
+`config.json` - it is a locked application manifest that Spin can use to run the
```

Comment:
for app deletion (and garbage collection), I wonder if we need an in text way of tying the config to the manifest, maybe adding the path to the manifest in the config? Or the way it is now, we can assume that the manifest is located in the same directory as the config

Reply:
As far as I can tell, you only need one of the manifest or locked application file (config) in order to do garbage collection, as they both are referring to the same layers / files on disk.

You would need to traverse all of them before marking something for garbage collection though, as layers can be shared across manifests / applications.